spec/spec_helper.rb

```diff
@@ -30,4 +30,10 @@
 
 RSpec.configure do |config|
   config.use_transactional_fixtures = true
+
+  # TODO: Remove once Devise publishes a new version
+  # https://github.com/heartcombo/devise/issues/5705
+  config.before do
+    Rails.application.try(:reload_routes_unless_loaded)
```

Comment:
@chaadow , I'm wondering if users will also need to add this to their test suite to make them happy 🤔

Reply:
only those on Rails 8 and above ( because of [lazy routes loading](https://github.com/rails/rails/pull/52353) ), but most of us that are on rails 8 use the "main/master" version of `devise` because they have not cut a release yet :/ 

because if you look at the changelog: https://github.com/heartcombo/devise/blob/main/CHANGELOG.md

everything is "well supported" but it's still not released yet. and since this gem tests against the most recent version of devise, I did this workaround to make the tests pass!

Your call @waiting-for-dev  🙏🏼

Reply:
Gotcha. How about this: we could put it behind a configuration option that's disabled by default. That way, we wouldn't break anyone's existing test suites. But folks who are aware of these quirks could enable it and add the call to their spec_helper file. What do you think? It'd also be good to add a notice in the README. Once Devise releases a new version, we can clean it up.

Reply:
By the way, for our test suite, we can probably just ` Rails.application.reload_routes_unless_loaded`, right? Given we're using Rails 8.

Reply:
@waiting-for-dev you are definitely right! I've just tested it and without it it fails. I also made a slight optimization to run it `before(:suite)`

Reply:
Hey @chaadow , just wanted to make sure you read this from above:

> How about this: we could put it behind a configuration option that's disabled by default. That way, we wouldn't break anyone's existing test suites. But folks who are aware of these quirks could enable it and add the call to their spec_helper file. What do you think? It'd also be good to add a notice in the README. Once Devise releases a new version, we can clean it up.

No hurries on my end, but I wouldn't like you to be left waiting for me